crates/cli-support/src/js/mod.rs

```diff
@@ -879,94 +701,89 @@ wasm = wasmInstance.exports;
+            // jsr-self-types directive
+            start.push_str(&format!(r#"/* @ts-self-types="./{module_name}.d.ts" */"#));
+            start.push('\n');
+        }
 
-        imports_init.push_str("imports[\"");
-        imports_init.push_str(&module_name);
-        imports_init.push_str("\"] = {};\n");
+        start.push_str(&format!(
+            r#"
+            import * as wasm from "./{module_name}_bg.wasm";
+            import {{ __wbg_set_wasm }} from "./{module_name}_bg.js";
+            __wbg_set_wasm(wasm);
+        "#
+        ));
 
-        for (id, js) in iter_by_import(&self.wasm_import_definitions, self.module) {
-            let import = self.module.imports.get_mut(*id);
-            import.module = module_name.clone();
-            imports_init.push_str("imports[\"");
-            imports_init.push_str(&module_name);
-            imports_init.push_str("\"].");
-            imports_init.push_str(&import.name);
-            imports_init.push_str(" = ");
-            imports_init.push_str(js.trim());
-            imports_init.push_str(";\n");
+        if needs_manual_start {
+            start.push_str("wasm.__wbindgen_start();");
         }
 
-        let extra_modules = self
-            .module
-            .imports
-            .iter()
-            .filter(|i| !self.wasm_import_definitions.contains_key(&i.id()))
-            .filter(|i| {
-                // Importing memory is handled specially in this area, so don't
-                // consider this a candidate for importing from extra modules.
-                !(matches!(i.kind, walrus::ImportKind::Memory(_)))
-            })
-            .map(|i| &i.module)
-            .collect::<BTreeSet<_>>();
-        for (i, extra) in extra_modules.iter().enumerate() {
-            let imports = match &mut imports {
-                Some(list) => list,
-                None => bail!("cannot import from modules (`{extra}`) with `--no-modules`"),
-            };
-            imports.push_str(&format!("import * as __wbg_star{i} from '{extra}';\n"));
-            imports_init.push_str(&format!("imports['{extra}'] = __wbg_star{i};\n"));
+        if !self.export_name_list.is_empty() {
+            start.push_str("\nexport {\n");
+            if let Some((last, list)) = self.export_name_list.split_last() {
+                for name in list {
+                    if is_valid_ident(name) {
+                        start.push_str(&format!("{name}, "));
+                    } else {
+                        start.push_str(&format!(r#""{name}", "#));
+                    }
+                }
+                if is_valid_ident(last) {
+                    start.push_str(last);
+                } else {
+                    start.push_str(&format!(r#""{last}""#));
+                }
+            }
+            start.push('\n');
+            start.push_str(r#"} from "./"#);
+            start.push_str(module_name);
+            start.push_str(r#"_bg.js";"#);
+            start.push('\n');
```

Comment:
Surely there should be a CommonJS path here? See also the missing `exported` in the reference CJS test.

Reply:
Got it, thanks.